### PR TITLE
processes: Add missing options

### PR DESCRIPTION
**Pull request by sileht**
processes: Add missing options

This change add some missing options:

* CollectContextSwitch
* CollectFileDescriptor
* CollectMemoryMaps

This options can be set globally or per process/processmatch.

### CODE REVIEW

**Comment by sileht** [comment]
@bastelfreak can you take a look ?

File: manifests/plugin/processes.pp [hunk 1]
```diff
@@ -5,6 +5,9 @@
   $interval                        = undef,
   Optional[Array] $processes       = undef,
   Optional[Array] $process_matches = undef,
+  Optional[Boolean] $collect_context_switch  = undef,
```
Comment (bastelfreak) [inline manifests/plugin/processes.pp]: yey datatypes \o/

Comment (bastelfreak) [reply manifests/plugin/processes.pp]: can you please align all = chars?

File: manifests/plugin/processes.pp [hunk 2]
```diff
@@ -25,21 +28,29 @@
   }
   concat::fragment { 'collectd_plugin_processes_conf_header':
     order   => '00',
-    content => '<Plugin processes>',
+    content => epp('collectd/plugin/processes-header.conf.epp'),
```
Comment (bastelfreak) [inline manifests/plugin/processes.pp]: epp \o/

File: manifests/plugin/processes/process.pp [hunk 3]
```diff
@@ -1,14 +1,22 @@
 define collectd::plugin::processes::process (
   $process = $name,
-  $ensure  = 'present'
+  $ensure  = 'present',
+  Optional[Boolean] $collect_context_switch  = undef,
```
Comment (bastelfreak) [inline manifests/plugin/processes/process.pp]: please align the = chars

Comment (sileht) [reply manifests/plugin/processes/process.pp]: Done

File: manifests/plugin/processes.pp (hunk 1 above)
Comment (sileht) [reply manifests/plugin/processes.pp]: Done

**Comment by bastelfreak** [comment]
Thanks @sileht !

**Pull request merged**